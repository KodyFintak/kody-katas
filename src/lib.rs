pub fn hello(name: &str) -> String {
    return format!("{} {}", "Hello", name);
}
