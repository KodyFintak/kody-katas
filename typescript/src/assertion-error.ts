export class AssertionError extends Error {
    name = "AssertionError";

    constructor(message: string, assertionFunction?: Function) {
        super(message);
        this.startStackTraceAt(assertionFunction);
    }

    private startStackTraceAt(assertionFunction?: Function) {
        if (Error.captureStackTrace && assertionFunction) {
            Error.captureStackTrace(this, assertionFunction);
            this.stack = this.stack?.split("\n").slice(0, 2).join("\n")
        }
    }
}