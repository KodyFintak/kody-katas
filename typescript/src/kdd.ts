export function it(name: string, test: () => void) {
    console.log(`- ${name}`)
    try {
        test();
        console.log("\tPASS")
    } catch (e: any) {
        console.error(`\tFAIL ${e.message}`)
    }
}

export function fail(message: string) {
    throw new Error(message);
}

export function expect<T>(actual: T) {
    return {
        toEqual(expected: T) {
            const testResult = isEqual(actual, expected);
            if (!testResult) fail(`expected ${JSON.stringify(actual)} === ${JSON.stringify(expected)}`)
        }
    }
}

function isEqual<T>(actual: T, expected: T) {
    if (typeof actual == "object") return objectsAreEqual(actual, expected);
    return actual === expected;
}

function objectsAreEqual<T>(actual: T, expected: T) {
    return JSON.stringify(actual) === JSON.stringify(expected);
}