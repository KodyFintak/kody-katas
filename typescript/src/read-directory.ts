import {readdirSync, statSync} from "fs";
import {join} from "path";

export function allFilesIn(directory: string): any[] {
    const files: any[] = [];

    const items = readdirSync(directory);

    for (const item of items) {
        const fullPath = join(directory, item);
        const stats = statSync(fullPath);

        if (stats.isDirectory()) {
            files.push(...allFilesIn(fullPath));
        }

        if (stats.isFile()) {
            files.push(fullPath);
        }
    }

    return files;
}