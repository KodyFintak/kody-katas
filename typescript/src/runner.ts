export class Runner {
    constructor(readonly tests: string[] = []) {
    }
    addTest(name: string) {
        this.tests.push(name)
    }
}

export const runner = new Runner()