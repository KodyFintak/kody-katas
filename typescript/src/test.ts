export class Test {
    constructor(public readonly name: string, private test: Function) {
    }

    run() {
        this.test()
    }
}