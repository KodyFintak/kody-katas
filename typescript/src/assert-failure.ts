import {fail} from "./kdd";

export function assertFailure(func: () => void) {
    try {
        func()
    } catch (e) {
        return
    }

    fail("expected failure, but did not get one")
}