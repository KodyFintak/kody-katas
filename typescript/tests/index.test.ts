import { GildedRose, Item } from '../src';

// kody, ellis, rob, sam, matt
describe('Gilded Rose Approval', () => {
    it('should foo', () => {
        const gildedRose = new GildedRose([new Item('foo2', 0, 0)]);
        const items = gildedRose.updateQuality();

        expect(items).toMatchSnapshot();
    });
});