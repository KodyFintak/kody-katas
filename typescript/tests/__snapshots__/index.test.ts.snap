// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Gilded Rose Approval > should foo 1`] = `
[
  Item {
    "name": "foo",
    "quality": 0,
    "sellIn": -1,
  },
]
`;
